frontend/src/components/dashboard/Analysis.tsx

```diff
@@ -104,39 +116,74 @@ const onLeaveLabel = (context: EventContext) => {
   return true;
 };
 
+const setAxis = (
```

Comment:
Since this function essentially manages internal state, with multiple disjoint `dispatch` calls, wouldn't it make sense for this to be handle by a single `dispatch`?

E.g.
```
dispatch({
  type: setAxis,
  {
    dimension: "x",
    type: newAxis,
  },
});
```
- `axisDimension` and `newAxis` naturally remain as parameters
- `axis` and `otherAxis` would already be part of the state, so no need to pass them as parameters.
- `splits` would already be part of the state, so no need to pass them as parameter.

Having this in a single dispatch would make the change more atomic - I don't see downsides 🤔

Reply:
Good catch - this makes it much cleaner! Somehow I have not noticed when writing, oops ^^